tasks/src/main/scala/dagr/tasks/parallel/PathBasedScatterGatherPipeline.scala

```diff
@@ -0,0 +1,142 @@
+    */
+  trait ScatterPathTask extends Task {
+    def output: Path
+  }
+
+  /** All tasks that gather multiple outputs into a single output should implement this task. */
+  trait GatherPathTask extends Task
+
+  /** A wrapper for the split input task. */
+  private class SplitInputPathAdapterTask(input: Path,
+                                          splitInputPathTaskGenerator: Path => SplitInputPathTask
+                                         ) extends SplitInputTask[Path, Path] {
+    def getTasks: Traversable[_ <: Task] = {
+      val splitInputTask = splitInputPathTaskGenerator(input)
+      val linker = Linker(from=splitInputTask, to=this)((from, to) => to._subDomains = Some(from.outputs))
+      splitInputTask ==> linker
+      List(splitInputTask, linker)
+    }
+  }
+
+  /** A wrapper for the scatter task. */
+  private class ScatterPathAdapterTask(input: Path,
+                                       scatterTaskGenerator: Path => ScatterPathTask,
+                                       deleteIntermediates: Boolean = true
+                                       ) extends ScatterTask[Path, Path] {
+    def getTasks: Traversable[_ <: Task] = {
+      val scatterTask = scatterTaskGenerator(input)
+      val linker = Linker(from=scatterTask, to=this)((from, to) => to._gatheredOutput = Some(from.output))
+      scatterTask ==> linker
+      if (deleteIntermediates) {
+        val deleteIntermediatesTask = new DeleteFiles(input)
+        scatterTask ==> deleteIntermediatesTask
+        List(scatterTask, linker, deleteIntermediatesTask)
+      }
+      else List(scatterTask, linker)
+    }
+  }
+
+  /** A wrapper for the gather task. */
+  private class GatherPathAdapterTask(inputs: Seq[Path],
+                                      output: Path,
+                                      gatherTaskGenerator: (Seq[Path], Path) => GatherPathTask,
+                                      deleteIntermediates: Boolean = true) extends GatherTask[Path] {
+    def gatheredOutput: Path = output
+    def getTasks: Traversable[_ <: Task] = {
+      val gatherTask = gatherTaskGenerator(inputs, output)
+      if (deleteIntermediates) {
+        val deleteIntermediatesTask = new DeleteFiles(inputs:_*)
+        gatherTask ==> deleteIntermediatesTask
+        List(gatherTask, deleteIntermediatesTask)
+      }
+      else List(gatherTask)
+    }
+  }
+}
+
+import dagr.tasks.parallel.PathBasedScatterGatherPipeline._
+
+/** A Scatter Gather Pipeline that operates on paths.  The input, intermediates, and output are all Paths.
+  *
+  * The constructor requires methods to generate the various task types.  The first path to the split input and
+  * scatter task are the temporary directory in which to operate, and the second argument is the input path.  The
+  * gather task takes a sequence of output paths to gather and the final output path.
+  */
+class PathBasedScatterGatherPipeline
+(
+  input: Path,
```

Comment:
I see why, but calling this `input` is confusing, in that the "input" to the HC pipeline is really the BAM, but from the S/G framework's perspective the BAM is irrelevant and the Intervals are what it's working on.  Maybe call this `domain`, or something along those lines?